apps/prairielearn/src/lib/externalGrader.ts

```diff
@@ -154,15 +154,15 @@ async function updateJobSubmissionTime(grading_job_id: string): Promise<void> {
     grading_job_id,
     grading_submitted_at: new Date().toISOString(),
   });
-  externalGradingSocket.gradingJobStatusUpdated(grading_job_id);
+  await externalGradingSocket.gradingJobStatusUpdated(grading_job_id);
```

Comment:
These calls were to a callback-based function but were not awaiting for the process to complete. I added await as a general rule to avoid floating promises, but if it is expected that this function should run in the background I can remove the awaits.